js/core/utils/array.js

```diff
@@ -3,10 +3,16 @@ import { orderEach } from './object';
 import config from '../config';
 
 function createOccurrenceMap(array) {
-    return array.reduce((map, value) => {
-        map[value] = (map[value] ?? 0) + 1;
-        return map;
-    }, {});
+    return array.reduce((map, value) => map.set(value, (map.get(value) ?? 0) + 1), new Map());
```

Comment:
It is really difficult to read. I think that readability is more important than shortness
```suggestion
    return array.reduce((map, value) => {
        const count = (map.get(value) ?? 0) + 1;
        map.set(value, count);
        return map;
    }, new Map());
```